go.mod

```diff
@@ -145,6 +145,7 @@ require (
 	github.com/gravitational/roundtrip v1.0.2
 	github.com/gravitational/teleport/api v0.0.0
 	github.com/gravitational/trace v1.5.1
+	github.com/gregjones/httpcache v0.0.0-20190611155906-901d90724c79
```

Comment:
Some library usage notes:
- `gregjones/httpcache` is deprecated and archived, however...
- We already include this library via the Kubernetes client library
- It otherwise appears to be fit for purpose, and no other similarly mature libraries exist ([`bartventer/httpcache`](https://github.com/bartventer/httpcache) exists but is much less mature)
- It is MIT licensed

I think we can switch to an alternative if/when we find a compelling reason to do so - it should just mean swapping out the implementation we return in `NewCachingHTTPClient()`.